### PR TITLE
Dataset generation

### PR DESCRIPTION
**Pull request by goastler**
Move dataset generation into typescript and fix fields to match what provider expects

### CODE REVIEW

**Review by forgetso — CHANGES_REQUESTED** [review]
Working now with most recent changes. The following items need addressed:

- [x] Linting
- [x] Language strings added to `@prosopo/common` for errors
- [x] Prosopo Error handling instead of default `Error`
```            
throw new Error(`Duplicate data entry in labelled data: ${JSON.stringify(entry)}`)
```
- [ ] Tests added to `packages/datasets/tests`

File: packages/datasets-fs/src/flatten/flatten.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { Args } from './args'
+import { CaptchaItemTypes, Data, DataSchema, LabelledItem } from '@prosopo/types'
+import { ProsopoEnvError } from '@prosopo/common'
+import { blake2b } from '@noble/hashes/blake2b'
+import { consola } from 'consola'
```
Comment (forgetso) [inline packages/datasets-fs/src/flatten/flatten.ts]: Switch to `import { logger } from @prosopo/common`

File: packages/datasets-fs/src/flatten/flatten.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { u8aToHex } from '@polkadot/util'
+import fs from 'fs'
+
+export default async (args: Args) => {
+    consola.log(args, 'flattening...')
```
Comment (forgetso) [inline packages/datasets-fs/src/flatten/flatten.ts]: Switch to `log = logger( ...etc); log.info`

File: packages/util/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "url": "git+https://github.com/prosopo/captcha.git"
+    },
+    "bugs": {
+        "url": "https://github.com/prosopo/captcha/issues"
+    },
+    "homepage": "https://github.com/prosopo/captcha#readme",
+    "publishConfig": {
+        "registry": "https://registry.npmjs.org"
+    },
+    "description": "Datasets for use with providers",
```
Comment (forgetso) [inline packages/util/package.json]: Update description

File: packages/util/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "author": "PROSOPO LIMITED <info@prosopo.io>",
+    "license": "Apache-2.0",
+    "private": false,
+    "scripts": {
+        "clean": "tsc --build --clean",
+        "build": "tsc --build --verbose",
+        "lint": "npx eslint .",
+        "lint:fix": "npx eslint . --fix",
+        "cli": "node ./dist/js/cli.js",
+        "test": "NODE_ENV=test ts-mocha \"./tests/js/**/*.test.ts\" --timeout 120000 --recursive --exit"
```
Comment (forgetso) [inline packages/util/package.json:13]: Do the tests work when run as part of the group test call from the root of the repo? `npm run test` from root

**Review by forgetso — CHANGES_REQUESTED** [review]

File: packages/util/package.json (hunk 4 above)
Comment (goastler) [reply packages/util/package.json:13]: there aren't any tests for this pkg, this is an artifact of copying/pasting the package.json from another package. Shall I remove this line?

File: packages/util/package.json (hunk 3 above)
Comment (goastler) [reply packages/util/package.json]: done

File: packages/datasets-fs/src/flatten/flatten.ts (hunk 1 above)
Comment (goastler) [reply packages/datasets-fs/src/flatten/flatten.ts]: done

File: packages/datasets-fs/src/flatten/flatten.ts (hunk 2 above)
Comment (goastler) [reply packages/datasets-fs/src/flatten/flatten.ts]: done

**Review by forgetso — DISMISSED** [review]

File: packages/common/package.json [hunk 5]
```diff
@@ -1,44 +1,44 @@
...
-    "lint:fix": "npx eslint . --fix"
-  },
-  "author": "",
-  "license": "ISC",
-  "dependencies": {
-    "@polkadot/keyring": "12.3.2",
-    "@polkadot/util": "12.3.2",
-    "@polkadot/util-crypto": "12.3.2",
-    "@types/react": "^18.0.25",
-    "consola": "^2.15.3",
```
Comment (forgetso) [inline packages/common/package.json:21]: Isn't `consola` still required?

**Review by forgetso — CHANGES_REQUESTED** [review]
Looks really good. Just one small query.

File: packages/common/package.json (hunk 5 above)
Comment (goastler) [reply packages/common/package.json:21]: erm yeah, not sure why that got removed. My bad!

**Review by forgetso — APPROVED** [review]

**Review by HughParry — APPROVED** [review]

**Pull request merged**